lib/mdns_lite/responder.ex

```diff
@@ -116,19 +116,21 @@ defmodule MdnsLite.Responder do
   def handle_info({:udp, _socket, src_ip, src_port, packet}, state) do
     # Decode the UDP packet
     with {:ok, dns_record} <- :inet_dns.decode(packet),
-         dns = DNS.Record.from_record(dns_record),
          # qr is the query/response flag; false (0) = query, true (1) = response
-         false <- dns.header.qr do
+         {:dns_rec, header, qdlist, _anlist, _nslist, _arlist} = dns_record,
```

Comment:
How about using Elixir's record support:

```elixir
require Record
Record.defrecord(:dns_rec, Record.extract(:dns_rec, from_lib: "kernel/src/inet_dns.hrl"))
```
(I just realized that this is pulling from `kernel/src` instead of `kernel/include` and this may fail completely.

Then this line would be:

```suggestion
         dns_rec(header: header, qdlist: qdlist) = dns_record,
```